### PR TITLE
feat: `WatchReceiver::wait_timeout`

### PR DESCRIPTION
**Pull request by dpc**
Life is too short to wait forever.

### CODE REVIEW

**Comment by dpc** [comment]
Seems like master branch is not up to date (missing v0.2.2)

File: src/sync_std.rs [hunk 1]
```diff
@@ -36,6 +36,16 @@ impl Condvar {
         }
     }
 
+    pub fn wait_timeout<'a, T>(
+        &self,
+        guard: MutexGuard<'a, T>,
+        duration: Duration,
+    ) -> Option<MutexGuard<'a, T>> {
+        match self.inner.wait_timeout(guard, duration) {
```
Comment (dpc) [inline src/sync_std.rs:44]: I was able to spot it hang indefinitely (with `duration` set to 1 second) - tested with `dbg!()` before and after and I have no idea why. The problem disappears when switching to `parking_lot`.

File: src/lib.rs [hunk 2]
```diff
@@ -154,6 +151,19 @@ impl<T: Clone> WatchReceiver<T> {
...
     }
+
+    /// This method waits until a new value becomes available and return a clone
+    /// of it, timing out after specified duration.
+    pub fn wait_timeout(&mut self, duration: Duration) -> Option<T> {
+        let mut lock = self.shared.lock.lock();
+
+        while lock.version == self.last_seen_version {
+            lock = self.shared.on_update.wait_timeout(lock, duration)?;
+        }
```
Comment (Darksonn) [inline src/lib.rs]: Every time this loop goes around, the deadline is reset. That's incorrect.

**Review by Darksonn — COMMENTED** [review]
I fixed the master branch.

File: src/lib.rs (hunk 2 above)
Comment (dpc) [reply src/lib.rs]: Right. Maybe that was my hang. Will fix.

**Review by Darksonn — APPROVED** [review]
LGTM, thanks.

**Pull request merged**